### PR TITLE
Main

### PR DESCRIPTION
**Pull request by LeahHarlow**

### CODE REVIEW

File: moveElementToEnd.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
+/* given an arrau and a target number, mutate the array in place and return the array with
```
Comment (rodcoelho) [inline moveElementToEnd.js]: nit: typo `arrau` should be `array`

Comment (LeahHarlow) [reply moveElementToEnd.js]: ty!

**Review by rodcoelho — COMMENTED** [review]
Great tests! And great initial pass.

Can we change the names of the variables to something that is easier on the eyes? Let's make `i` something like `iter` or `element` and let's make `j` something like `lenOfArray` or `len`. This will make it easier to revisit this code in the future.

Let's also add some comments where we might be explaining what we're doing as if we were talking to the interviewer.

Last thing, once you've done the above: How can we improve the main function? It seems like we iterate over the whole array when we should probably stop once we found the one we're looking for.

**Comment by LeahHarlow** [comment]
Thank you for the feedback! I made the variable names clearer and added comments to clarify the thought process. The reason it iterates over the whole array is because there can be multiples and we want to move all of the target numbers to the end of the array!

**Review by rodcoelho — APPROVED** [review]
LGTM, good catch on moving multiple elements to the end - I didn't read the prompt thoroughly, I mainly focused on the unit tests you wrote.

One last nit: put the comments in the line before the code rather than the end of the line. This is the convention I've seen normally. Other than that let's merge it! 🚀

**Pull request merged**